zipline/data/data_portal.py

```diff
@@ -311,6 +323,146 @@ def _get_daily_data(self, asset_int, column, dt):
         else:
             return raw_value
 
+    def _get_history_daily_window(self, sids, end_dt, bar_count, field_to_use):
+        """
+        Internal method that returns a dataframe containing history bars
+        of daily frequency for the given sids.
+        """
+        data = []
```

Comment:
Our daily window logic: if the last day is partial, using minute data to calculate the value of the partial day.  also, creating an extra (empty) slot in the np array that is reserved for the partial day, so that we don't have to copy the array once we have daily data and need to add the partial data.

We could/should switch to USEquityPricingLoader and figure out how to maintain those details, but not important enough right now.  After shipping.